src/middlewares/createContextMiddleware.ts

```diff
@@ -128,7 +128,43 @@ export const createContextMiddleware = ({ config }: Props) => {
       return members;
     };
 
+    const fetchRemoteMembersMentionList = async (
+      silenced: boolean = false
+    ) => {
+      const allMembers = ctx.database.getAllRemoteMembers();
+
+      const membersFetchResult = await Promise.allSettled(
+        Object.keys(allMembers).map(async (userId) => {
+          try {
+            const member = await ctx.getChatMember(Number(userId));
+            return createMemberMention(member.user, silenced);
+          } catch (err: any) {
+            if (
+              err.code === 400 &&
+              err.message.includes('user not found')
+            ) {
+              ctx.logger.warn(
+                `Registered user with id "${userId}" does not exist. Removing user remote.`
+              );
+              ctx.database.removeRemoteMember(Number(userId));
+            }
+            throw err;
+          }
+        })
```

Comment:
probably would be nice to have this function extracted into it's own:

```ts
const fetchMemberMention = async (userId: number, silenced: boolean) => {
  try {
    const member = await ctx.getChatMember(Number(userId));
    return createMemberMention(member.user, silenced);
  } catch (err: any) {
    if (
      err.code === 400 &&
      err.message.includes('user not found')
    ) {
      ctx.logger.warn(
        `Registered user with id "${userId}" does not exist. Removing user remote.`
      );
      ctx.database.removeRemoteMember(Number(userId));
    }
    throw err;
  }
}
```

and then just use it in the `fetchMembersMentionList` and `fetchRemoteMembersMentionList`:

```ts
// fetchMembersMentionList version
const membersFetchResult = await Promise.allSettled(
  memberIds.map((userId) => {
    return fetchMemberMention(userId, silenced)
  })
);

// fetchRemoteMembersMentionList version
const membersFetchResult = await Promise.allSettled(
  Object.keys(allMembers).map((userId) => {
    return fetchMemberMention(userId, silenced)
  })
);
```

Reply:
Nice idea, will implement it !

Reply:
Done.